### PR TITLE
Convert stubborn JSON values to usable types

### PR DESCRIPTION
**Pull request by nexdrew**
Initial changes to support JSON modification via converting one key at a time to a more usable data type and printing the modified content to stdout, which can be redirected to write to file.

Here's an example (old output from trackpipe):

``` sh
$ hewer cgw-12k.log -k analytics
Mining JSON file: cgw-12k.log
Total Rows: 3
JSON Rows: 3
Key: analytics
Total Encounters: 3
```

What? Where are subkeys? (Look at file.) Oh, `analytics` is a JSON-encoded string. Ok, let's do this:

``` sh
$ hewer cgw-12k.log -k analytics -c json > cgw-12k-converted.log
$ hewer cgw-12k-converted.log -k analytics
Mining JSON file: cgw-12k-converted.log
Total Rows: 3
JSON Rows: 3
Key: analytics
Total Encounters: 3
Subkeys: [activeTrackers activitySeconds packetLoss responses runtimeSeconds tracks]
```

That's better. Now let's try `analytics.packetLoss.percentage`:

``` sh
$ hewer cgw-12k-converted.log -k analytics.packetLoss.percentage
Mining JSON file: cgw-12k-converted.log
Total Rows: 3
JSON Rows: 3
Key: analytics.packetLoss.percentage
Total Encounters: 3
```

Well, crap. (Look in file again.) Oh, those values are strings. Take this:

``` sh
$ hewer cgw-12k-converted.log -k analytics.packetLoss.percentage -c number > cgw-12k-usable.log
$ hewer cgw-12k-usable.log -k analytics.packetLoss.percentage
Mining JSON file: cgw-12k-usable.log
Total Rows: 3
JSON Rows: 3
Key: analytics.packetLoss.percentage
Total Encounters: 3
Average: 73
Max: 80
Min: 62
```

Ah, there we go.

Also bumped the minor version since stuff was added to the CLI.

``` sh
$ hewer -v
0.2.0
```

### CODE REVIEW

File: miner.go [hunk 1]
```diff
@@ -38,3 +38,29 @@ func ParseFile(fileName string, analytics *Analytics) {
 		fmt.Printf("Skipped %d non-JSON lines in '%s'.\n", skippedLines, fileName)
 	}
 }
+
+func ParseAndConvertFile(fileName string, converter *Converter) {
```
Comment (booee) [inline miner.go:42]: In a later change, we could align the `Converter` and `Analtyics` interfaces, so we wouldn't need to create two separate methods

**Comment by booee** [comment]
I like the idea of cleaning the log files :+1: 
Piping the output to new files is awesome :+1: 

But why manipulate the files at all? This requires digging into a potentially large log file full of JSON content to do your own troubleshooting (multiple times via your example, even though that JSON was super-ugly). This requires intimate knowledge of _hewer_ and how it works/interprets lines. Instead, we could just implicitly do conversions without the user's knowledge, print the output, and [business as usual](https://www.youtube.com/watch?v=dQw4w9WgXcQ). A bit less imperative, but the user wouldn't have to get their hands dirty

Merge at will

File: miner.go (hunk 1 above)
Comment (nexdrew) [inline miner.go:42]: I thought of this, but I didn't take the time to figure out how to create an interface in Go. I figured I would wait to see what you thought before I went that deep.

**Comment by nexdrew** [comment]
I agree it would be slick to do this automagically behind the scenes, but we'd have to do a series of checks/parse attempts for each string value encountered and then do multiple passes if any JSON-encoded strings are converted to JSON. In case there are legitimate string values (like ids, error messages, etc), this would be very inefficient. Plus the modified file may be useful for other programs/intentions as well, so it seemed best to leave the power in the user's hands.

I don't think this really requires any special knowledge of how hewer works - any JSON parser is gonna have the same problems with the same content. At any rate, the user probably has some idea of what the file contents look like, and so this just allows them to tell hewer what needs to be done to it.

Perhaps if hewer could aggregate/summarize _all_ keys in the file in one pass, it would make more sense to handle conversions automatically, but until then, this is a workable alternative.

**Comment by booee** [comment]
Complicated? Yes, probably requiring a new class just to encapsulating the coercion, among other caveats here and there that would need to be handled

Inefficient? Probably. At least in the context of re-running a bad file multiple times while and not preserving the conversions

OK, carry on

**Pull request merged**